Wip pg provision script and diagram (#394)
ipfs-shipyard/nft.storage: __WIP_pg_schema_review/readme.md

> # If you only read one thing, read the `provision.sql` file

Reviewer: I would make a note that it's generated from .prima file

Author: yes I just copied it at the root, I'll actually make a convenience script for this

# SQL Migrataion scratchpad Repo

This is a repo to test out some ideas for migrating and defining the sql schemas in the postgres database that will be the target for the migration away from fauna

At this moment (8/30/2021) there's just a few folders

> - niftysave- Literally just a copy of the niftysave repo so its nearby and convenient to look at

Reviewer: This is gone now

```suggestion
```

Author: sorry this should be deleted, I forgot to remove it on my new branch

Author: yes I'll move it into the databases section

- prisma-and-gql-test - right now its empty but I plan to test a nexus endpoint on top of the generated prisma client to see how difficult it is

> - prisma-test - this is where I've worked most recently and its got the schemas and migrations generated from prisma

Reviewer: I think above two bullet points our outdated now.

## Quick Start

1. Make sure you have docker installed and working, im running a postgres image there atm
2. cd into the `prisma-test` and `docker-compose up` should start the image. I've exposted the port on 5439 so it doesn't clash with other things on my system but you're welcome to change this.
3. Be aware the connection string is in the .env file. This is just a test to its `jondoe`, `randompassword` for now.
4. `npm install`. The Prisma directory has 3 things

>    - **dbml**. Every time we run `npx prisma generate` a new .dbml file is made. You can paste this at `https://dbdiagram.io/d` and check out the visualizer

Reviewer: ```suggestion
   - **dbml**. Every time we run `npm gen:dbml` a new `.dbml` file is made. You can paste this at `https://dbdiagram.io/d` and check out the visualizer
```

>    - **migrations**. This is where every named migration lives. Once we've made a good schema we can put thes in a .old folder and jsut make the one init migration but in the meantime you can observe the sort of sql prisma emits here

Reviewer: Seems outdated

```suggestion
```

   - **schema.prisma**. The actual Prisma schema!

The commands you care about most are:

> 1. `npx prisma migrate dev` : will create a named migration, and you can go look at the sql

Reviewer: I think we're leaving it out for now

```suggestion
1. `npx prisma migrate dev` : will create a named migration, and you can go look at the sql
```
